apps/prairielearn/src/lib/code-caller/index.ts

```diff
@@ -50,27 +49,33 @@ export async function init() {
   }
 
   const numWorkers = config.workersCount ?? Math.ceil(config.workersPerCpu * os.cpus().length);
-  pool = createPool(
+  pool = createPool<CodeCaller>(
     {
       create: async () => {
-        let codeCallerOptions = {
+        const codeCallerOptions = {
           questionTimeoutMilliseconds: config.questionTimeoutMilliseconds,
           pingTimeoutMilliseconds: config.workerPingTimeoutMilliseconds,
           errorLogger: logger.error.bind(logger),
         };
-        let codeCaller;
-        if (workersExecutionMode === 'container') {
-          codeCaller = new CodeCallerContainer(codeCallerOptions);
-        } else if (workersExecutionMode === 'native') {
-          codeCaller = new CodeCallerNative(codeCallerOptions);
-        } else {
-          throw new Error(`Unexpected workersExecutionMode: ${workersExecutionMode}`);
-        }
+
+        const codeCaller = run(() => {
+          if (workersExecutionMode === 'container') {
+            return new CodeCallerContainer(codeCallerOptions);
+          } else if (workersExecutionMode === 'native') {
+            return new CodeCallerNative(codeCallerOptions);
+          } else {
+            throw new Error(`Unexpected workersExecutionMode: ${workersExecutionMode}`);
+          }
+        });
+
         await codeCaller.ensureChild();
         load.startJob('python_worker_idle', codeCaller.uuid);
         return codeCaller;
       },
       destroy: async (codeCaller) => {
+        logger.info(
+          `Destroying Python worker ${codeCaller.uuid} (last course path: ${codeCaller.getCoursePath()})`,
+        );
```

Comment:
This is the only behavior change. This is a bit of a stab in the dark to try to gain more information about why the code caller queue sometimes gets backed up; based on other system logs, it seems like code callers are being killed.